content/articles/oop-in-java-abstraction-and-encapsulation/index.md

```diff
@@ -0,0 +1,456 @@
+ 
+Here's another example to make things clearer: imagine car drivers. They are only interested in starting, moving, and stopping the car, speed, or changing gear and brake. They are not interested in the mechanism that allows them to do this. In abstraction, we hide the implementation details from the user while exposing the functionality.
+ 
+This way, the user will only know what a function (method in java) does but not how it does it.
+ 
+### How to implement Abstraction in Java
+In Java, abstraction is implemented using either of these two ways:
+- By using the non-access modifier *abstract* keyword or
+- Interface
+ 
+#### Working with the Abstract Keyword
+You can only use the abstract access modifier keyword with classes and methods but not with variables. Methods with the keyword *abstract* are methods without implementation.
+ 
+To implement an abstract class in Java, use the abstract keyword, e.g., ```public abstract class Circle```, where Circle is the class name. Let's illustrate this using this code snippet:
+ 
+```java
+ 
+public abstract class Dog {
+ 
+ public abstract void run();
+ 
+}
+ 
+```
+ 
+With the above code snippet, we've created an abstract class and an abstract method. One pertinent feature of abstract class is that objects cannot be instantiated or created from it. But don’t forget, classes only exist to create an object. To be able to create an object from an abstract class, we must create a new class that inherits from the superclass (base/parent).You can do this using the ```extends``` keyword.
+ 
+Here is a Code snippet to illustrate the ```extends``` keyword:
+ 
+```java
+ 
+public class MainDog extends Dog {
+ 
+ //abstract method body definition
+ 
+ public void run(){
+ 
+     //in this class the method is implemented
+ 
+     System.out.println(“Dog is running”);
+ 
+ }
+ 
+ public static void main(Strings [] args){
+ 
+     MainDog dog = new MainDog();
+ 
+     //abstract method invocation
+ 
+     dog.run();
+ 
+ }
+ 
+}
+ 
+```
+ 
+#### Interface implementation
+An interface contains public, abstract method signatures (methods without implementation). It is also important to note that interfaces have no constructors and may contain final and static variables. Implementating abstraction by the interface method is possible by the use of the `implements` keyword.
+The child class inherits the parent class, overrides its method signatures, and provides a method body.
+ 
+```java
+ 
+public interface Car{
+ 
+ public void start();
+ 
+ public void stop();
+ 
+}
+ 
+public Benz implements Car{
+ 
+ /*@Override to override the stop method in the car class that has no implementation*/
```

Comment:
This comment makes it sound as though the annotation is required to override a method when that is not true. It's very minor so it's probably not that big of a deal but I would consider changing this comment to take that into account.

Reply:
Done